### PR TITLE
IDEX-2726: consider dev machine as running only after successful star…

### PR DESCRIPTION
**Pull request by garagatyi**
…t of ws agent
@skabashnyuk @evoevodin

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ок

**Comment by akorneta** [comment]
ok

**Comment by skabashnyuk** [comment]
ок

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncher.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.machine.server;
+
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.machine.server.exception.MachineException;
+
+/**
+ * Starts ws agent in the machine and wait until ws agent sends notification about its start
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncher.java]: I think the sentence is not correct one, you should use eihter `Starts ws agent ... and waits ...` or `Start ws agent ... and  wait ...`

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncher.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * @author Alexander Garagatyi
+ */
+public interface WsAgentLauncher {
+    void startWsAgent(String workspaceId) throws NotFoundException, MachineException, InterruptedException;
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncher.java]: Is it necessary to have javadoc for this method?

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncherImpl.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        this.wsAgentPingPath = apiEndpoint.getPath();
+    }
+
+    @Override
+    public void startWsAgent(String workspaceId) throws NotFoundException, MachineException, InterruptedException {
+        final Instance devMachine = machineManager.getDevMachine(workspaceId);
+        try {
+            machineManager.exec(devMachine.getId(),
+                                new CommandImpl(WS_AGENT_PROCESS_NAME, wsAgentStartCommandLine, "Arbitrary"),
+                                String.format(WS_AGENT_PROCESS_OUTPUT_CHANNEL, workspaceId));
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncherImpl.java]: I think in this case it is better to have `public static` method for getting channel.
Why:
The modifier of the `WS_AGENT_PROCESS_OUTPUT_CHANNEL` is public, which means that you want to reuse it in the other place, so you will do the same operation there `String.format(WS_AGENT_PROCESS_OUTPUT_CHANNEL, workspaceId)`.  If you create the method for it the usage will be obvious and won't let the developer to miss something

**Comment by evoevodin** [comment]
Ok

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncher.java (hunk 1 above)
Comment (garagatyi) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncher.java]: yes

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncher.java (hunk 2 above)
Comment (garagatyi) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncher.java]: It is interface with one method, so javadoc for interface explains everything. Is it not enough in your opinion?

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncherImpl.java (hunk 3 above)
Comment (garagatyi) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncherImpl.java]: +1

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncher.java (hunk 2 above)
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/WsAgentLauncher.java]: it is okay =)

**Pull request merged**